source/generate_cids/generate_concept_dicts.py

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Encoding required to deal with 'micro' character
+"""Script for auto-generating DICOM SR context groups from FHIR JSON value set
+resources.
+
+
+"""
+
+import ftplib
+import json
+import logging
+import os
+import re
+from io import BytesIO
+import tempfile
+import glob
+
+logger = logging.getLogger(__name__)
+
+# Example excerpt fhir JSON for reference
+"""
+    "resourceType":"ValueSet",
+    "id":"dicom-cid-10-InterventionalDrug",
+    ...
+    "name":"InterventionalDrug",
+    ...
+    "compose":{
+        "include":[
+            {
+                "system":"http://snomed.info/srt",
+                "concept":[
+                    {
+                        "code":"F-B2135",
+                        "display":"Epinephrine"
+                    },
+
+"""
+# The list of scheme designators is not complete.
+# For full list see table 8-1 in part 3.16 chapter 8:
+# http://dicom.nema.org/medical/dicom/current/output/chtml/part16/chapter_8.html#table_8-1
+FHIR_SYSTEM_TO_DICOM_SCHEME_DESIGNATOR = {
+    'http://snomed.info/sct': 'SCT',
+    'http://snomed.info/srt': 'SRT',
+    'http://dicom.nema.org/resources/ontology/DCM': 'DCM',
+    'http://loinc.org': 'LN',
+    'http://www.radlex.org': 'RADLEX',
+    'http://sig.biostr.washington.edu/projects/fm/AboutFM.html': 'FMA',
+    'http://www.nlm.nih.gov/mesh/meshhome.html': 'MSH',
+    'http://ncit.nci.nih.gov': 'NCIt',
+    'http://unitsofmeasure.org': 'UCUM',
+    'http://hl7.org/fhir/sid/ndc': 'NDC',
+    'urn:iso:std:iso:11073:10101': 'MDC',
+    'doi:10.1016/S0735-1097(99)00126-6': 'BARI',
+    'http://www.nlm.nih.gov/research/umls': 'UMLS',
+    'http://pubchem.ncbi.nlm.nih.gov': 'PUBCHEM_CID',
+    'http://braininfo.rprc.washington.edu/aboutBrainInfo.aspx#NeuroNames': 'NEU',
+    'http://www.itis.gov': 'ITIS_TSN',
+}
+
+leave_alone = ['mm', 'cm', 'km', 'um',
+               'ms',  #  'us'?-doesn't seem to be there
+               'ml',
+               'mg', 'kg',
+               ]  # ... probably need others
+
+def camel_case(s):
+    return ''.join(word.capitalize() if word != word.upper()
+                   and word not in leave_alone
+                   else word
+                   for word in re.split(r"\W", s, flags=re.UNICODE)
```

Comment:
I believe this function is problematic, since it will drop all non-alphanumeric characters, resulting, for example, in both "T2" and "T2*" mapping to "T2", and leading to duplicate items in the dictionary.

Reply:
I made a PR to add a warning: https://github.com/hackermd/pydicom/pull/1

Reply:
But I think the issue should be fixed for those situations where we know this function fails, like the T2* example. Do you agree?